ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -469,28 +490,40 @@ public class BottomSheetController: UIViewController {
         bottomSheetOffsetConstraint.constant = -offsetFromBottom
     }
 
-    // MARK: - Height constraint utils
+    private func updateSheetSizingConstraints() {
+        if preferredExpandedContentHeight > 0 {
+            fullScreenSheetConstraint.isActive = false
 
-    private func updateBottomSheetHeightConstraint() {
-        let newConstraint = generateBottomSheetHeightConstraint()
+            // Apply the new preferred height to the static layout guide
+            preferredExpandedContentGuideTopConstraint?.constant = -preferredExpandedContentHeight
 
-        bottomSheetHeightConstraint.isActive = false
-        newConstraint.isActive = true
+            // Activate constraint which ties expandedContentView height to the layout guide height
+            preferredExpandedContentHeightConstraint.isActive = true
 
-        bottomSheetHeightConstraint = newConstraint
+        } else {
+            // Tie the sheet size to maxSheetHeightLayoutGuide.heightAnchor to make it full screen
+            preferredExpandedContentHeightConstraint.isActive = false
+            fullScreenSheetConstraint.isActive = true
+        }
     }
 
-    private func generateBottomSheetHeightConstraint() -> NSLayoutConstraint {
-        // Fill view bounds, respecting the given height fraction
-        let constraint = bottomSheetView.heightAnchor.constraint(
-            equalTo: view.heightAnchor,
-            multiplier: expandedHeightFraction,
-            constant: view.safeAreaInsets.top - Constants.minimumTopExpandedPadding)
+    private lazy var preferredExpandedContentHeightConstraint: NSLayoutConstraint = {
+        let constraint = expandedContentView.heightAnchor.constraint(equalTo: preferredExpandedContentLayoutGuide.heightAnchor)
+        constraint.priority = .defaultHigh
```

Comment:
Any time we have a non-required constraint that is implicit ambiguity. It may be the right call (I think it is in this case), but is always worth a comment

Reply:
Yeah all the height constraints have to be lower than required so AutoLayout can automatically enforce the max height constraint.

Reply:
could you add a comment here as well?